app/src/main/java/org/glucosio/android/presenter/A1CCalculatorPresenter.java

```diff
@@ -72,7 +72,14 @@ public void checkGlucoseUnit() {
     }
 
     public void saveA1C(double a1c) {
-        HB1ACReading a1cReading = new HB1ACReading(a1c, new Date());
+        User user = dbHandler.getUser(1);
+        double finalA1c = a1c;
+        if (!"percentage".equals(user.getPreferred_unit_a1c())) {
+            GlucosioConverter converter = new GlucosioConverter();
+            finalA1c = converter.a1cIfccToNgsp(a1c);
```

Comment:
BTW this is explained in Converter's comments.
`    public double a1cNgspToIfcc(double ngsp) {
        // percentage to mmol/mol
        // [NGSP - 2.152] / 0.09148
        return round((ngsp - 2.152) / 0.09148, 2);
    }

    public double a1cIfccToNgsp(double ifcc) {
        // mmol/mol to percentage
        // [0.09148 * IFCC] + 2.152
        return round((0.09148 * ifcc) + 2.152, 2);
    }`